### PR TITLE
pkp/pkp-lib#6894 Allow selecting Chapter files when chapter created

### PR DESCRIPTION
**Pull request by ajnyga**

### CODE REVIEW

**Comment by ajnyga** [comment]
Some tests failed but I do not think it was because of the changes, restart those? @asmecher

**Comment by ajnyga** [comment]
Thanks, they passed!

File: controllers/grid/users/chapter/form/ChapterForm.inc.php [hunk 1]
```diff
@@ -157,31 +160,33 @@ function fetch($request, $template = null, $display = false) {
...
 				}
 			}
-			$templateMgr = TemplateManager::getManager($request);
-			$templateMgr->assign([
-				'chapterFileOptions' => $chapterFileOptions,
-				'selectedChapterFiles' => $selectedChapterFiles,
-			]);
 		}
 
+		$submissionFiles = Services::get('submissionFile')->getMany(['submissionIds' => [$this->getMonograph()->getId()]]);
```
Comment (NateWr) [inline controllers/grid/users/chapter/form/ChapterForm.inc.php:177]: It looks like this duplicates a call to get the files at line 169. Can you get the submission files once and loop over them once to get the `chapterFileOptions` and the `selectedChapterFiles`?

**Review by NateWr — CHANGES_REQUESTED** [review]
:+1: I left one comment in there to prevent a duplicate DB lookup, but otherwise looks good.

File: controllers/grid/users/chapter/form/ChapterForm.inc.php (hunk 1 above)
Comment (ajnyga) [reply controllers/grid/users/chapter/form/ChapterForm.inc.php:177]: Yeah sure, probably that is possible. That seemed like a clean way of doing it, but you are right that it causes unnecessary queries which we already have plenty.

**Comment by ajnyga** [comment]
Question: should we actually allow authors to attach files to specific chapters during the submission process?

**Comment by NateWr** [comment]
> Question: should we actually allow authors to attach files to specific chapters during the submission process?

I think so. I can't see any reason why not and it may help editors sort through a submission with many chapter files.

**Comment by ajnyga** [comment]
What is a bit problematic in OMP is that there is no clear distinction between a chapter manuscript file and a published chapter file. Meaning that you can, to my knowledge, attach a file that has been sent in the submission stage to a published chapter.

Imo there should be a more clear difference between the files that are being published and the files that are submitted as book components in the beginning.

I can not see any real life situation where a publisher would want to publish the first version of a chapter that is most likely in some other format than pdf/html.

**Comment by ajnyga** [comment]
@NateWr did a new pr and started tests.

Other thing that came to mind is that it would be nice if the Genre of the file would be visible as well.

**Review by NateWr — APPROVED** [review]

**Comment by NateWr** [comment]
Looks good. Let me know when the tests have passed and I can merge this in.

**Comment by ajnyga** [comment]
@NateWr just one test failed, can you restart

**Comment by asmecher** [comment]
(I restarted the test)

**Comment by ajnyga** [comment]
thanks, tests are passing!

**Pull request merged**